Cilsil/Cil/Parsers/ExceptionParser.cs

```diff
@@ -0,0 +1,311 @@
+
+                // Handles loaded exception expression in finally block.
+                var exceptionVariable = new LvarExpression(
+                                            new LocalVariable(Identifier.ByteCodeIdentifier,
+                                                            state.Method));
+                state.PushExpr(exceptionVariable, new Tstruct("System.Object"));
+
+                state.PushInstruction(instruction);
+            }
+
+            // Append the next instruction (which should be stloc, for representing
+            // storage of the constructed object into a local variable) to this new node.
+            (instruction, _) = state.PopInstruction();
+            ParseCilInstruction(instruction, state);
+
+            return true; 
+        }
+
+        /// <summary>
+        /// Check if there is a finally block following the current catch block.
+        /// </summary>
+        /// <param name="state">Current program state.</param>
+        /// <param name="catchEndOffset">The ending offset of catch block.</param>
+        private bool ContainsFinallyBlock(ProgramState state, int catchEndOffset)
+        {
+            return state.ExceptionBlockStartToEndOffsets.ContainsKey(catchEndOffset + 3)
+                   || state.ExceptionBlockStartToEndOffsets.ContainsKey(catchEndOffset + 2);
+        }
+
+        /// <summary>
+        /// Creates a exception method call returned via out parameter.
+        /// </summary>
+        /// <param name="state">Current program state.</param>
+        /// <param name="exceptedExpression">The excepted expression.</param>
+        /// <param name="exceptedExpressionType">The type of excepted expression.</param>
+        /// <param name="returnType">The return type of the exception method being called.</param>
+        /// <param name="returnVariable">Identifies the variable returned by the method.</param>
+        /// <param name="callArgs">The method arguments.</param>
+        /// <param name="exceptionCall">The exception Call SIL instruction.</param>
+        private void CreateExceptionCall(ProgramState state,
+                                         Expression exceptedExpression,
+                                         Typ exceptedExpressionType,
+                                         out TypeReference returnType,
+                                         out Identifier returnVariable,
+                                         out List<Call.CallArg> callArgs,
+                                         out Call exceptionCall)
+        {
+            callArgs = new List<Call.CallArg>();
+            try
+            {
+                returnType = state.OffsetToExceptionType[state.CurrentInstruction.Offset];
+            }   
+            catch(KeyNotFoundException)
+            {
+                returnType = state.Method.Module.TypeSystem.Object;
+            }
+            
+            var funcExp = new ConstExpression(ProcedureName.BuiltIn__unwrap_exception);
```

Comment:
I'm a bit confused -- why are we putting unwrap exception in the frontend but not reflecting it in the backend? https://github.com/xi-liu-ds/infer/blob/8b78964b497705f86ae7c4fc06edc63b938e2184/infer/src/backend/InferAnalyzeJson.ml#L70

Reply:
I think we commented this out because we failed one example long before. Do you still have that example? Because I didn't see it breaks anything for now.